src/sql/src/util.rs

```diff
@@ -21,8 +21,8 @@ use sqlparser::ast::{ObjectName, SqlOption, Value};
 
 static SQL_SECRET_PATTERNS: LazyLock<Vec<Regex>> = LazyLock::new(|| {
     vec![
-        Regex::new(r#"(?i)access_key_id=["'](\w*)["'].*"#).unwrap(),
-        Regex::new(r#"(?i)secret_access_key=["'](\w*)["'].*"#).unwrap(),
+        Regex::new(r#"(?i)access_key_id=["']([^"']*)["'].*"#).unwrap(),
```

Comment:
This would not cover the key contains `\"` but anyway we redact the whole secrets.

Syntax error can cause the matching brittle so we may handle syntax error in other way.

Reply:
I believe this could be a temporary solution; we may discover a better method in the future.

Reply:
Yeah. IIRC ClickHouse would use AST everywhere and reconstruct the SQL string when needed. With a more structural object we can do checks / processes more preciously.